libparsec/crates/platform_storage/tests/unit/certificates.rs

```diff
@@ -798,3 +798,52 @@ async fn forget_all_certificates(mut timestamps: TimestampGenerator, env: &Testb
     )
     .await;
 }
+
+#[parsec_test(testbed = "minimal")]
+async fn prohibit_skipping_indexes(mut timestamps: TimestampGenerator, env: &TestbedEnv) {
+    let alice = env.local_device("alice@dev1");
+
+    let storage = CertificatesStorage::start(&env.discriminant_dir, &alice)
+        .await
+        .unwrap();
+
+    let t1 = timestamps.next();
+    let data = AddCertificateData::from_user_certificate(
+        b"<encrypted>".to_vec(),
+        t1,
+        alice.device_id.user_id().clone(),
+    );
+
+    // Try with no certificates skipping 1 index
+    storage.add_next_certificate(2, data).await.unwrap_err();
```

Comment:
Must check error using a `p_assert_matches!`, otherwise an unexpected error can shadow the one we expect (this happened before :laughing: )